presto-kinesis/src/test/java/io/prestosql/plugin/kinesis/util/MockKinesisClient.java

```diff
@@ -0,0 +1,604 @@
+                result = new GetRecordsResult();
+                List<Record> recs = shard.getRecords();
+                result.setRecords(recs); // NOTE: getting all for now
+                result.setNextShardIterator(getNextShardIterator(iter, recs).makeString());
+                result.setMillisBehindLatest(100L);
+            }
+            else {
+                result = new GetRecordsResult();
+                List<Record> recs = shard.getRecordsFrom(iter);
+                result.setRecords(recs); // may be empty
+                result.setNextShardIterator(getNextShardIterator(iter, recs).makeString());
+                result.setMillisBehindLatest(100L);
+            }
+        }
+        else {
+            throw new AmazonClientException("Unknown stream or bad shard iterator.");
+        }
+
+        return result;
+    }
+
+    protected ShardIterator getNextShardIterator(ShardIterator previousIter, List<Record> records)
+    {
+        ShardIterator newIter = null;
+        if (records.size() == 0) {
+            newIter = previousIter;
+        }
+        else {
+            Record rec = records.get(records.size() - 1);
+            int lastSeq = Integer.valueOf(rec.getSequenceNumber());
+            newIter = new ShardIterator(previousIter.streamId, previousIter.shardIndex, lastSeq + 1);
+        }
+
+        return newIter;
+    }
+
+    //// Unsupported methods
+
+    @Override
+    public ListTagsForStreamResult listTagsForStream(ListTagsForStreamRequest listTagsForStreamRequest)
+            throws AmazonClientException
+    {
+        return null;
+    }
+
+    @Override
+    public ListStreamsResult listStreams(ListStreamsRequest listStreamsRequest)
+            throws AmazonClientException
+    {
+        return null;
+    }
+
+    @Override
+    public ListStreamsResult listStreams()
+            throws AmazonServiceException, AmazonClientException
+    {
+        return null;
+    }
+
+    @Override
+    public PutRecordResult putRecord(String s, ByteBuffer byteBuffer, String s1)
+            throws AmazonServiceException, AmazonClientException
+    {
+        throw new UnsupportedOperationException("MockKinesisClient doesn't support this.");
+    }
+
+    @Override
+    public PutRecordResult putRecord(String s, ByteBuffer byteBuffer, String s1, String s2)
+            throws AmazonServiceException, AmazonClientException
+    {
+        throw new UnsupportedOperationException("MockKinesisClient doesn't support this.");
+    }
+
+    @Override
+    public DescribeStreamResult describeStream(String s)
+            throws AmazonServiceException, AmazonClientException
+    {
+        return null;
+    }
+
+    @Override
+    public DescribeStreamResult describeStream(String s, String s1)
+            throws AmazonServiceException, AmazonClientException
+    {
+        return null;
+    }
+
+    @Override
+    public DescribeStreamResult describeStream(String s, Integer integer, String s1)
```

Comment:
Please fix the parameter names for all of these methods, as these names are not descriptive.

Reply:
changed